### DISCUSSION TITLE
Problem with installation under Windows 10

### DISCUSSION THREAD

**Discussion by fedemone** [mliebelt/pgn-viewer#429] [Q&A] [answered]
Hi,

let me say that I was interested in using the PgnViewerJS for a Discourse based forum, so I wanted to install locally and try to create some new BBcode/markdown to use the PGN.
I tried basic installation "npm install @mliebelt/pgn-viewer" and got error: "'$npm_execpath' is not recognized as an internal or external command,".

I tried to set it as environment variable, to reinstall the node.js (especially in path with spaces) and other tricks (I checked also where the command using $npm_execpath is created and tried to hack it a bit hardcoding some paths), but nope. 
I googled about this problem or similar, but got no luck in so far.
I'm attaching the logs, does anybody there have an idea of what's going wrong here?

[2023-04-03T08_52_44_524Z-debug-0.log](https://github.com/mliebelt/PgnViewerJS/files/11138318/2023-04-03T08_52_44_524Z-debug-0.log)
[2023-04-03T08_52_52_641Z-debug-0.log](https://github.com/mliebelt/PgnViewerJS/files/11138319/2023-04-03T08_52_52_641Z-debug-0.log)

**Comment by fedemone** [comment, marked as answer]
It seems that installing yarn (`npm install --global yarn` and `yarn add yarpm`) and using that instead (`yarn add @mliebelt/pgn-reader`, `yarn add @mliebelt/pgn-viewer`), the reader/viewer has been installed locally successfully.

**Reply by mliebelt** [reply]
The question here is, what "installed locally" means. In the normal context of doing web development, and wanting to use `pgnv` (or any other function), you will have to `import` it somehow in your context. Then `import ...` will do the job for you. If you want to use it in you HTML/CSS context (as part of a web page), you have to find the one file `dist.js.gz` and load it as library in your web page.

So more concrete examples in the `README.md` would help, but I cannot tell what you scenario at the end will look like. So thank you for solving your problem yourself, but I have not understood what is _wrong_ in the setup of `pgn-viewer`. Btw `pgn-viewer` includes completely `pgn-reader`, so this one file is then sufficient to use it.

**Reply by mliebelt** [reply]
I looked at the logs, and the part you were referencing in the beginning is strange. It contains some more errors, and the one error at the end was one from `Chessground` that is only used as a library. Strange.

Tried to do it locally, and of course is succeeds. Builds this structure then.
```~/tmp/ch1 » l
total 24K
drwxrwxr-x 18 mliebelt mliebelt 4,0K Apr  3 16:27 node_modules/
-rw-rw-r--  1 mliebelt mliebelt   65 Apr  3 16:27 package.json
-rw-rw-r--  1 mliebelt mliebelt  13K Apr  3 16:27 package-lock.json

~/tmp/ch1 » ll node_modules/@mliebelt/pgn-viewer/lib
total 2804
-rw-rw-r-- 1 mliebelt mliebelt  949467 Apr  3 16:27 dist.js
-rw-rw-r-- 1 mliebelt mliebelt  329393 Apr  3 16:27 dist.js.gz
-rw-rw-r-- 1 mliebelt mliebelt 1280717 Apr  3 16:27 dist.js.map
-rw-rw-r-- 1 mliebelt mliebelt  206580 Apr  3 16:27 dist.js.map.gz
...

**Reply by fedemone** [reply]
Thanks a lot for your reply. I must admit that I'm an ignorant of JS (I usually work with C/C++ and Perl/Python), so I did not try some deep debug of the problem, and probably - since there's a completely valid workaround - there's no need. Possibly the installation is missing some some environment variable, or it's simply unset somewhere, and yarn is able to retrieve it / keep it stored (I was trying to read the module code but did not spend too much time on that).
About the "locally" meaning, sorry for being unclear, some phrasing can be misleading. I'm working on a normal laptop in order to customize the Discourse theme, adding the javascript (I'm reading carefully your guide!) so to have a proper handling of PGN in a HTML page formatted in Discourse fashion. Then I'll upload the theme to the server, enable it for private use and I'll verify the work so far. Let's see if I am able to do that cleanly.

I did some basic comparison, and result with yarn is perfect. Maybe you want to add that for an alternative compile method. 
Again, thank you so much for everything.
